src/test/kotlin/app/batch/HttpWriterTest.kt

```diff
@@ -1,121 +1,200 @@
+        val decryptedStream = DecryptedStream(ByteArrayInputStream(byteArray), filename, "$s3Path/$filename")
         val httpClient = Mockito.mock(CloseableHttpClient::class.java)
         given(httpClientProvider.client()).willReturn(httpClient)
         val httpResponse = Mockito.mock(CloseableHttpResponse::class.java)
-        given(httpClient.execute(ArgumentMatchers.any(HttpPost::class.java))).willReturn(httpResponse)
+        given(httpClient.execute(any(HttpPost::class.java))).willReturn(httpResponse)
         val statusLine = Mockito.mock(StatusLine::class.java)
         given(statusLine.statusCode).willReturn(200)
         given(httpResponse.statusLine).willReturn((statusLine))
+
+        //when
         httpWriter.write(mutableListOf(decryptedStream));
-        Mockito.verify(httpClient, Mockito.times(1))
-                .execute(ArgumentMatchers.any(HttpPost::class.java))
+
+        //then
+        val httpCaptor = argumentCaptor<HttpPost>()
+        verify(httpClient, once()).execute(httpCaptor.capture())
+        assertEquals("Content-Type: application/octet-stream", httpCaptor.firstValue.entity.contentType.toString())
+        assertEquals(2, httpCaptor.firstValue.allHeaders.size)
+        assertEquals("filename: db.core.addressDeclaration-000001.txt.bz2.enc", httpCaptor.firstValue.allHeaders[0].toString())
+        assertEquals("collection: db.core.addressDeclaration", httpCaptor.firstValue.allHeaders[1].toString())
+
+        verify(mockS3StatusFileWriter, once()).writeStatus(decryptedStream.fullPath)
+
+        val logCaptor = argumentCaptor<ILoggingEvent>()
+        verify(mockAppender, Mockito.times(5)).doAppend(logCaptor.capture())
+        val formattedMessages = logCaptor.allValues.map { it.formattedMessage }
+        assertEquals("Writing: '1' items", formattedMessages[0])
+        assertEquals("Checking: 'exporter-output/job01/db.core.addressDeclaration-000001.txt.bz2.enc'", formattedMessages[1])
+        assertEquals("Found collection: 'db.core.addressDeclaration' from fileName of 'exporter-output/job01/db.core.addressDeclaration-000001.txt.bz2.enc'", formattedMessages[2])
+        assertEquals("Posting: 'exporter-output/job01/db.core.addressDeclaration-000001.txt.bz2.enc' to 'db.core.addressDeclaration'", formattedMessages[3])
+        assertEquals("Successfully posted 'exporter-output/job01/db.core.addressDeclaration-000001.txt.bz2.enc': response '200'", formattedMessages[4])
     }
 
     @Test
-    fun testAlternateFilenamePattern() {
-        val byteArray = "hello, world".toByteArray()
+    fun test_will_write_to_nifi_when_valid_file_with_embedded_hyphens_in_dbname() {
         val filename = "db.core-with-hyphen.addressDeclaration-000001.txt.bz2.enc"
-        val decryptedStream = DecryptedStream(ByteArrayInputStream(byteArray), filename)
+        val decryptedStream = DecryptedStream(ByteArrayInputStream(byteArray), filename, "$s3Path/$filename")
```

Comment:
seems strange that filename is here twice. 
Is this the definition: `data class DecryptedStream(val inputStream: InputStream, val filename: String)`?
This occurs in several places, but I won't comment those

Reply:
We need the filename. (for nifi) and the full path (for the S3 status writer)

Reply:
That definition is from master :-)